pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -295,6 +297,7 @@ func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandb
 			}
 			return msg, ErrPostStartHook
 		}
+		metrics.ContainerPostStartHookDuration.Observe(time.Since(postHookStartTime).Seconds())
```

Comment:
Do you intentionally emit this metrics only when the PostStart hook succeeds?

Reply:
yes, the duration on a failed one does not make sense and shouldn't be looked at.